src/sentry/security/utils.py

```diff
@@ -7,16 +7,18 @@
 from .emails import generate_security_email
 
 if TYPE_CHECKING:
-    from sentry.models import User
+    from django.contrib.auth.models import AnonymousUser
+
+    from sentry.models import AbstractBaseUser
 
 
 logger = logging.getLogger("sentry.security")
 
 
 def capture_security_activity(
-    account: "User",
+    account: "AbstractBaseUser",
     type: str,
-    actor: "User",
+    actor: "AbstractBaseUser | AnonymousUser",
```

Comment:
I adjusted/extended these types as this was failing.

For one, most places that use this pass in `request.user` as the actor, which can be AnonymousUser. So I figure this is actually correct here.

For the other part, it was complaining about passing in a RcpUser for account (from org.get_owners()). Changing this to AbstractBaseUser worked, for whatever reason - TBH I don't know a lot about python inheritance and type checking, but this seems to satisfy it at least. cc @iambriccardo

Reply:
so, should I make it `AbstractBaseUser | AnonymousUser | RpcUser` then? 

Related question for understanding, out of scope for this PR, but is there a reason that RpcUser and User have no shared base class/interface (e.g. why RpcUser does not implement AbstractBaseUser)? Just to better wrap my head around this :)